core/src/smartcontracts/wasm.rs

```diff
@@ -746,12 +811,104 @@ impl RuntimeBuilder {
+                Runtime::$fn_ident,
+            )
+            $(.and_then(|l| {
+                l.func_wrap(
+                    import::MODULE_NAME,
+                    import::$multi_name,
+                    Runtime::$multi_fn_ident,
+                )
+            }))*
+            .map_err(|err| Error::Initialization(eyre!(Box::new(err))))
+    };
+}
+
+impl<'wrld> RuntimeBuilder<state::SmartContract<'wrld>> {
+    /// Builds the [`Runtime`] for *Smart Contract* execution
+    ///
+    /// # Errors
+    ///
+    /// Fails if failed to create default linker.
+    pub fn build(self) -> Result<Runtime<state::SmartContract<'wrld>>> {
+        self.finalize(|engine| {
+            let mut linker = Linker::new(engine);
+
+            create_imports!(linker,
+                import::EXECUTE_ISI_FN_NAME => execute_instruction,
+                import::EXECUTE_QUERY_FN_NAME => execute_query,
+                import::QUERY_AUTHORITY_FN_NAME => query_authority,
+                import::QUERY_MAX_LOG_LEVEL => query_max_log_level,
+                import::LOG_FN_NAME => log,
+                import::DBG_FN_NAME => dbg,
+            )?;
+            Ok(linker)
+        })
+    }
+}
+
+impl<'wrld> RuntimeBuilder<state::Trigger<'wrld>> {
+    /// Builds the [`Runtime`] for *Trigger* execution
+    ///
+    /// # Errors
+    ///
+    /// Fails if failed to create default linker.
+    pub fn build(self) -> Result<Runtime<state::Trigger<'wrld>>> {
+        self.finalize(|engine| {
+            let mut linker = Linker::new(engine);
+
+            create_imports!(linker,
+                import::EXECUTE_ISI_FN_NAME => execute_instruction,
+                import::EXECUTE_QUERY_FN_NAME => execute_query,
+                import::QUERY_AUTHORITY_FN_NAME => query_authority,
+                import::QUERY_MAX_LOG_LEVEL => query_max_log_level,
+                import::QUERY_TRIGGERING_EVENT_FN_NAME => query_triggering_event,
+                import::LOG_FN_NAME => log,
+                import::DBG_FN_NAME => dbg,
+            )?;
+            Ok(linker)
+        })
+    }
+}
+
+impl<'wrld> RuntimeBuilder<state::Validator<'wrld>> {
+    /// Builds the [`Runtime`] for *Validator* execution
+    ///
+    /// # Errors
+    ///
+    /// Fails if failed to create default linker.
+    pub fn build(self) -> Result<Runtime<state::Validator<'wrld>>> {
+        self.finalize(|engine| {
+            let mut linker = Linker::new(engine);
+
+            create_imports!(linker,
+                import::EXECUTE_ISI_FN_NAME => execute_instruction_as_validator,
+                import::EXECUTE_QUERY_FN_NAME => execute_query_as_validator,
+                import::QUERY_AUTHORITY_FN_NAME => query_authority,
+                import::QUERY_OPERATION_TO_VALIDATE_FN_NAME => query_operation_to_validate,
+                import::QUERY_MAX_LOG_LEVEL => query_max_log_level,
+                import::LOG_FN_NAME => log,
+                import::DBG_FN_NAME => dbg,
```

Comment:
```suggestion
                import::EXECUTE_ISI_FN_NAME => execute_instruction_as_validator,
                import::EXECUTE_QUERY_FN_NAME => execute_query_as_validator,
                import::QUERY_OPERATION_TO_VALIDATE_FN_NAME => query_operation_to_validate,
                import::QUERY_MAX_LOG_LEVEL => query_max_log_level,
```

these should always be defined, so you can just hardcode them in the macro

Reply:
I wasn't referring to those functions